crates/kitsune_p2p/kitsune_p2p/src/config.rs

```diff
@@ -0,0 +1,73 @@
+use url2::Url2;
+
+/// Configure the kitsune actor
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, PartialEq)]
+pub struct KitsuneP2pConfig {
+    /// list of sub-transports to be included in this pool
+    pub transport_pool: Vec<TransportConfig>,
+}
+
+/// Configure the network bindings for underlying kitsune transports
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, PartialEq)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum TransportConfig {
+    /// A transport that uses the QUIC protocol
+    Quic {
+        /// To which network interface / port should we bind?
+        /// Default: "kitsune-quic://0.0.0.0:0".
+        bind_to: Option<Url2>,
+
+        /// If you have port-forwarding set up,
+        /// or wish to apply a vanity domain name,
+        /// you may need to override the local NIC ip.
+        /// Default: None = use NIC ip.
+        override_host: Option<String>,
+
+        /// If you have port-forwarding set up,
+        /// you may need to override the local NIC port.
+        /// Default: None = use NIC port.
+        override_port: Option<u16>,
+    },
+    /// A transport that tls tunnels through a sub-transport (ALPN kitsune-proxy/0)
+    Proxy {
+        /// The 'Proxy' transport is a wrapper around a sub-transport
+        /// We also need to define the sub-transport.
+        sub_transport: Box<TransportConfig>,
+
+        /// Determines whether we wish to:
+        /// - proxy through a remove
```

Comment:
little spelling error could lead to confusion... :p 
"/// - proxy through a remove" -> "/// - proxy through a remote"

Reply:
```suggestion
        /// - proxy through a remote
```

Haha, thanks @Connoropolous !